js/lex.go

```diff
@@ -161,6 +161,9 @@ func (l *Lexer) Next() (TokenType, []byte) {
 	// ErrorToken, WhitespaceToken and CommentToken are already returned
 	if tt == LineTerminatorToken || tt == PunctuatorToken && regexpStateByte[c] {
 		l.regexpState = true
+	} else if tt == IdentifierToken {
+		hash := ToHash(l.r.Lexeme())
+		l.regexpState = hash != 0 && hash != This
```

Comment:
Oh, I didn't notice `true` and `false` are in the hash table, and from the first glance just assumed it contains for keywords only.

For others this is intended, as it's simpler to check this way. This is part of Sweet.js algorithm I mentioned earlier, which relies on, as I said in the description:

> if you see a keyword, and it's not `this`, it can be followed only by either regular expression or be a syntax error

Now that I look more carefully and see that `null`, `false`, `true` are all there as well, there are two ways of dealing with it - either add few more items to the comparison (maybe by making it a `switch` so that it gets nicely optimized), or yeah, make a table, although that one will be harder to maintain over time IMO while bringing not much benefits.